nodejs/week4/spam-detector/index.js

```diff
@@ -0,0 +1,89 @@
+class Email {
+	constructor(subject, body) {
+		this.subject = subject;
+		this.body = body;
+	}
+}
+
+class SpamDetector {
+	badWordCheck(email) {
+		const badWords = ["viagra", "offer", "free", "business proposal"];
+
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split(" ");
+
+		const badWordsInEmail = emailBodyIntoArray.filter((item) => {
+			return badWords.includes(item.toLowerCase());
+		});
+
+		if (badWordsInEmail.length !== 0) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	subjectCheck(email) {
+		const emailSubject = email.subject;
+
+		if (emailSubject.toLowerCase() === "hello") {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	upperCaseCheck(email) {
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split("");
+		const upperCaseArray = emailBodyIntoArray.filter((words) => {
```

Comment:
be very careful with these names, they can cheat if you write words people will think array

```suggestion
		const upperCaseArray = emailBodyIntoArray.filter((word) => {
```

Reply:
okay, but what do you mean by they can cheat?

Reply:
I mean confuse not cheat. If another developer sees words he will expect its an array, but its actually an object